shell/platform/linux/fl_keyboard_manager.cc

```diff
@@ -479,8 +480,35 @@ FlKeyboardManager* fl_keyboard_manager_new(
         } else {
           g_autoptr(FlEngine) engine = FL_ENGINE(g_weak_ref_get(&self->engine));
           if (engine != nullptr) {
-            fl_engine_send_key_event(engine, event, callback,
-                                     callback_user_data);
+            typedef struct {
+              FlutterKeyEventCallback callback;
+              void* callback_user_data;
+            } SendKeyEventData;
+            SendKeyEventData* data = g_new0(SendKeyEventData, 1);
+            data->callback = callback;
+            data->callback_user_data = callback_user_data;
+            fl_engine_send_key_event(
+                engine, event, self->cancellable,
+                [](GObject* object, GAsyncResult* result, gpointer user_data) {
+                  g_autofree SendKeyEventData* data =
+                      static_cast<SendKeyEventData*>(user_data);
+                  gboolean handled = FALSE;
+                  g_autoptr(GError) error = nullptr;
+                  if (!fl_engine_send_key_event_finish(
```

Comment:
This is the way that GObject does asynchronous calls, the purpose of this change is to make all the code more consistent and reliable (in particular by using GCancellable to make sure all async calls are correctly aborted if the caller is disposed before a call completes). I'm not sure the history of why this pattern was chosen. Some ideas of why it might not be desirable to do the finish internally:
- This would require making custom callbacks for each function, which might be considered harder to write and have difficulty with language bindings.
- The finish completes the async call, which might do some specific behaviour. The callback only indicates the function is ready to finish, you might store the `GAsyncResult` and do it later or chain it into another async call.